src/pykeen/evaluation/rank_based_evaluator.py

```diff
@@ -474,53 +475,45 @@ def process_scores_(
         self.num_entities = num_entities
 
 
+@lru_cache(maxsize=3)
+def _get_key(target: Target) -> List[TargetColumn]:
```

Comment:
you could just pre-instantiate a dictionary with these values in it

```
KEY_THING = {
    target: [TARGET_TO_INDEX[c] for c in COLUMN_LABELS if c != target]
    for target in COLUMN_LABELS
}

Reply:
Yes, https://github.com/pykeen/pykeen/pull/1079/commits/a5cfee31bf7bdeacb5988cc7a9c88247af12a6a1